src/main/java/lotto/domain/Lotto.java

```diff
@@ -0,0 +1,60 @@
+package lotto.domain;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+public class Lotto {
+
+    private static final int LOTTO_SIZE = 6;
+    private final List<LottoNumber> numbers;
+
+    protected Lotto(List<LottoNumber> numbers) {
+        validateLottoSize(numbers);
+        this.numbers = numbers;
+    }
+
+    private void validateLottoSize(List<LottoNumber> numbers) {
+        if (numbers.size() != LOTTO_SIZE) {
+            throw new IllegalArgumentException("로또 번호는 6개여야 합니다.");
+        }
+    }
+
+    public static Lotto create(LottoNumberStrategy lottoNumberStrategy) {
+        return new Lotto(lottoNumberStrategy.create());
+    }
+
+    public Rank match(Lotto winningLotto) {
+        int matchCount = 0;
+        Set<LottoNumber> collect = new HashSet<>(winningLotto.numbers);
+
+        for (LottoNumber number : this.numbers) {
+            if (collect.contains(number)) {
+                matchCount++;
+            }
+        }
```

Comment:
```
프로그래밍 요구사항
- indent(인덴트, 들여쓰기) depth를 2를 넘지 않도록 구현한다. 1까지만 허용한다.
    - 예를 들어 while문 안에 if문이 있으면 들여쓰기는 2이다.
    - 힌트: indent(인덴트, 들여쓰기) depth를 줄이는 좋은 방법은 함수(또는 메소드)를 분리하면 된다.
```

다음과 같은 프로그래밍 요구사항이 존재하는데요,
어떻게 해결해 볼 수 있을까요? 😄

Reply:
프로그래밍 요구사항을 생각하지 않고 구현하였네요 :(
힌트에서 주신것 처럼 메소드로 분리하여 할 수도 있을 것 같지만 stream에 익숙해지기 위해 스트림을 한번 사용해 봤습니다.
```suggestion
        int matchCount = (int) this.numbers.stream()
                .filter(lottoNumber -> collect.contains(lottoNumber))
                .count();
```